app/views/visual_groups/edit.html.erb

```diff
@@ -0,0 +1,52 @@
+<%
+  @title = :edit_visual_group_title.t(name: @visual_group.name)
+%>
+
+<p id="notice"><%= notice %></p>
+
+<p>
+  <%= link_to(:visual_group_show.t, @visual_group) %> |
+  <%= link_to(:visual_group_back_to_index.t,
+              visual_model_visual_groups_path(@visual_group.visual_model)) %>
+</p>
+
+<%= render 'form', visual_group: @visual_group %>
+
+<hr>
+
+<p>
+  <strong><%= :visual_group_includes_names.t %>:</strong><br>
+  <% @visual_group.distinct_names.each do |name| %>
+  <%= link_to(name[0],
+              edit_visual_group_path(@visual_group, filter: name[0]))%><br>
+  <% end %>
+</p>
+
+<hr>
+
+<strong><%= :edit_visual_group_filter_options.t %></strong>
+<%= form_with(url: edit_visual_group_path(@visual_group), method: :get) do |form| %>
+  <%= form.text_field(:filter, value: @filter, size: 60) %>
+  <p>
+    <% ["needs_review", "included", "excluded"].each do |a_status| %>
+      <%= form.radio_button(:status, a_status, checked: @status == a_status, id: a_status) %>
+      <%= form.label(:"status_#{a_status}", :"visual_group_#{a_status}".t, value: a_status, for: a_status) %>
+    <% end %>
+  </p>
+
+  <p>
+    <div class="actions">
+      <%= form.submit(:edit_visual_group_update_filter.t) %>
+    </div>
+  </p>
```

Comment:
I think you could use a simple inline JS, something like this (i'm not sure I have the actual HTML id's right):

```suggestion
  </p>
  <script>
    jQuery("#included, #excluded").on("click", function(e){
    // or jQuery('input[name="status[included]"], input[name="status[excluded]"]').on("change click", function(e){
      jQuery("#filter").val("");
    });
  </script>
```

Reply:
Thanks!  That helped.  Needed a bit more tweaking to get it to work, but it seems to be working now.  I also added an anchor to avoid bouncing back to the top on a small screen.  I like @pellaea's idea of disabling the text box as well when "Included" or "Excluded" are selected, but I'll need to figure out how to do that.

Reply:
Now that I've played with it a bit, it seems like it should go even further and do the reload when you click "Included" or "Excluded".  Then the button could be renamed "Review" and be next to the text box.  I'm also thinking this really should be it's own thing and the stuff for editing the name and description should be on it's own page.

Reply:
Hit a bug.  If you use the links to mark something as "Included" or "Excluded" under the images and then click "Update Filter" (with idea being that it should remove the images you've provided info for), the "Update Filter" button becomes disabled.  This isn't anything I did consciously so it's a bit of a head scratcher for me.  Any suggestions welcome!

Reply:
@mo-nathan  - rails-ujs probably disables the buttons on submit. The `data-disable-with` attribute is automatically added to the submit button. (...and you can change the text of that button while disabled. I also made a `safe_spinner` helper method that you can use to add a spinner after whatever text you use.)

There's a rails-ujs way to re-enable a disabled button with your jQuery snippet, on page load, or on whatever event.
```
setTimeout(function(){
  document.querySelectorAll('[data-disable-with]').forEach(function(element){
    $.rails.enableElement(element);
  });
});
```

Reply:
Sorry, I need more help on this.  JavaScript just doesn't make any sense to me.  I don't know how to get it to run that code so it makes any difference.  I just seeing the exact same behavior.

Reply:
Actually, in this case, maybe it would be better to remove the data-disable-with attribute from these buttons?  I wonder if there is an option you can pass into form.submit() that does this?  If there were some way like that to avoid having it disable the button in the first place, that would seem like a far cleaner and simpler solution.

However, in answer, perhaps, to your question about how to run the javascript, you would need to add an event handler to the submit button that runs the setTimeout(...) snippet @nimmolo supplied.  Something like this:

<script>jQuery("#<submit_button_id>").on<Event>(function() { setTimeout(...) })</script>

I forget which event you would want to attach it to, and of course fill in the correct button id.  Also, I'm not sure if you want to run this immediately, or place it all in a document.onLoad() callback to have it run after the page has loaded!  Maybe @nimmolo knows these details off of the top of his head?  Also, don't you have to add a 1 second delay or something to the setTimeout() call?

Reply:
Decided to mostly bail on that approach and just go with links rather than radio buttons and provide a reload button.  The "Update Filter" button can still get stuck if you click the links under the images and then click "Update Filter" without change the name, but if you hit reload it's recovers.  Not sure it's worth spending more time on at this point.

Reply:
@mo-nathan @pellaea  I had forgotten how i'd already dealt with this on #1160.
Made a suggestion in the code review (small addition to `main.js`) that will re-enable the button if any inputs change.

You could also disable `data-disable-with` but I'd much prefer to keep it on and get submit-disabling working properly on the forms - then we don't need that alert telling people not to submit twice, we simply disable it.

Reply:
I realize this is a dead horse, because Nathan's taking a different approach now, but I just want to understand the issue better...

@nimmolo -- Are you saying that we have to explicitly pass an option into the submit button to make it disable when you press it?   Are there cases where we don't want that behavior?  Maybe this is one of those cases?  Or if, say, it is calling an AJAX method, it might make sense to have the AJAX callback(s) re-enable the button, so that even if the user doesn't change an input field they can still press the button again after the first press finishes?  Are we on more or less the same page?  Thanks!

Reply:
Yep i did say that and sorry — that was sloppy and incorrect advice.

I eventually went back to my PR and review what i learned (more than a month ago)! 
Editing that comment now.

Reply:
Also - yes, AJAX callbacks could re-enable submit buttons using the same bit of JS.